mm2src/coins/lp_coins.rs

```diff
@@ -1575,6 +1642,38 @@ pub trait MakerCoinSwapOpsV2: CoinAssocTypes + Send + Sync + 'static {
     async fn spend_maker_payment_v2(&self, args: SpendMakerPaymentArgs<'_, Self>) -> Result<Self::Tx, TransactionErr>;
 }
 
+#[async_trait]
+pub trait MakerNftSwapOpsV2: CoinAssocTypes + NftAssocTypes + Send + Sync + 'static {
```

Comment:
Same comment, are you planning to make a different `maker_swap_v2.rs` and `taker_swap_v2.rs` files for NFTs?

Reply:
No, I wanted only to have NFT related states in `maker_swap_v2.rs` and `taker_swap_v2.rs` for `MakerSwapStateMachine` and `TakerSwapStateMachine`.
Well yeah for nft states we will need to require NFT related trait impl here `impl<MakerCoin: MmCoin + MakerCoinSwapOpsV2, TakerCoin: MmCoin + TakerCoinSwapOpsV2> State`

But we will discuss the nft swap approach of course.